hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncBatchRpcRetryingCaller.java

```diff
@@ -486,6 +487,9 @@ private void tryResubmit(Stream<Action> actions, int tries, boolean immediately,
     } else {
       delayNs = getPauseTime(pauseNsToUse, tries - 1);
     }
+    Optional<MetricsConnection> metrics = conn.getConnectionMetrics();
+    // TODO(baugenreich) should this only report when the server is overloaded
```

Comment:
In general is it best to report this only when the server is overloaded? Or are we interested in the backoff time always?

Reply:
The original goal of the issue is to be able to track how much time a client is backing off specific to overloaded server. The back off time is typically going to be relative to how much load a client is sending (so the problematic client hammering a server will hopefully be the one experiencing the most back off)

It might be interesting to report back off time overall, but in that case I think we should have two separate timers -- 1 for general, and 1 for overloaded.